File: Questions/Diamond_Pattern/problem.c
#include <stdio.h>

void print_diamond_pattern(int n) {
    // Upper half of the diamond
    for (int i = 1; i <= n; i += 2) {
        int spaces = (n - i) / 2 + 1;
        for (int s = 0; s < spaces; s++) {
            printf(" ");
        }
        for (int j = 0; j < i; j++) {
            printf("%d", i - j);
        }
        printf("\n");
    }

    // Lower half of the diamond
    for (int i = n - 2; i > 0; i -= 2) {
        int spaces = (n - i) / 2;
        for (int s = 0; s < spaces; s++) {
            printf(" ");
        }
        for (int j = 0; j < i; j++) {
            printf("%d", i + j);
        }
        printf("\n");
    }
}

int main() {
    // Example usage
    print_diamond_pattern(5);
    return 0;
}

File: Questions/Diamond_Pattern/problem.py
# Difficulty: Hard

def print_diamond_pattern(n):
    # Upper half of the diamond
    for i in range(1, n + 1, 2):
        spaces = " " * ((n - i) // 2 + 1)
        numbers = "".join(str(i - j) for j in range(i)) 
        print(spaces + numbers)

    # Lower half of the diamond
    for i in range(n - 2, 0, 2):
        spaces = " " * ((n - i) // 2)
        numbers = "".join(str(i + j) for j in range(i))
        print(spaces + numbers)

# Answer:
# def print_diamond_pattern(n):
#     # Upper half of the diamond
#     for i in range(1, n + 1, 2):
#         spaces = " " * ((n - i) // 2) # Incorrect calculation of spaces, should be (n - i) // 2
#         numbers = "".join(str(i + j) for j in range(i)) # Incorrect calculation of numbers, should be (i + j)
#         print(spaces + numbers) 

#     # Lower half of the diamond
#     for i in range(n - 2, 0, -2): # Incorrect range, should be (n - 2, 0, -2)
#         spaces = " " * ((n - i) // 2)
#         numbers = "".join(str(i + j) for j in range(i))
#         print(spaces + numbers)


# Sample Test Cases
print_diamond_pattern(5)